### ISSUE TITLE
Overly verbose WebSocket logging

### ISSUE THREAD

**Issue by snej** [couchbase/couchbase-lite-android#1546]
The amount of WebSocket logging is making it very hard to read Android replication logs, and it's probably slowing down replication too. For every Sync log there are hundreds of lines like:
```
E/WebSocket: WebSocketListener.onMessage() bytes -> 180100c3b512d4
E/C4Socket: C4Socket.callback.write() handle -> 0xf095c32c, allocatedData.length -> 57
E/C4Socket: C4Socket.callback.completedReceive() socket -> 0xf095c32c, byteCount -> 7
E/WebSocket: WebSocketListener.onMessage() bytes -> 200100537f4c52
E/C4Socket: C4Socket.callback.completedReceive() socket -> 0xf095c32c, byteCount -> 7
E/WebSocket: WebSocketListener.onMessage() bytes -> 2201005734a019
E/C4Socket: C4Socket.callback.completedReceive() socket -> 0xf095c32c, byteCount -> 7
E/C4Socket: C4Socket.callback.write() handle -> 0xf095c32c, allocatedData.length -> 57
E/C4Socket: C4Socket.callback.write() handle -> 0xf095c32c, allocatedData.length -> 56
E/C4Socket: C4Socket.callback.write() handle -> 0xf095c32c, allocatedData.length -> 57
E/C4Socket: C4Socket.callback.write() handle -> 0xf095c32c, allocatedData.length -> 58
E/C4Socket: C4Socket.callback.write() handle -> 0xf095c32c, allocatedData.length -> 56
```
These are logged at Error level so they always appear. If it's necessary to keep these, could you lower them to Verbose?

* The `WebSocketListener` messages appear to come from shared/src/main/java/com/couchbase/lite/internal/replicator/CBLWebSocket.java
* The `C4Socket` messages are in LiteCore, at Java/src/com/couchbase/litecore/C4Socket.java

**Comment by snej** [comment]
The C4Socket messages were earlier reported in couchbase/couchbase-lite-core#343

**Comment by snej** [comment]
Shouldn't these be logged to C4Log, not the Android logger?

**Comment by hideki** [comment]
Hi @snej,

The current implementation uses Android logger for Java codes. And the native code logging uses C4Log, but it also prints logs with C android logging library to the `logcat`.

The reason Java code does not use C4Log is to avoid JNI call and logging target is the `logcat`. If JNI overhead is small, we could switch to C4Log.

There are two reasons to avoid logging callback from native to Java:
- Callback from native to Java middle of copying data with using the critical mode, such as `GetPrimitiveArrayCritical` and `GetStringCritical`, causes a crash. None critical mode does not cause, but we prefer to use critical mode for performance. I found this during we worked for `cbforest`.
- Calling Java method from native is slow in general.
